### DISCUSSION TITLE
How do I make types created using the newtype pattern work?

### DISCUSSION THREAD

**Discussion by tasn** [SeaQL/sea-orm#477] [Q&A] [answered]
Example code:
```
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "application")]
pub struct Model {
    pub my_type: MyType,
}

pub struct MyType(pub String);
```
How can I make this type work (as a string)?

I tried doing something like this:
```
impl Into<sea_orm::Value> for MyType {
    fn into(self) -> sea_orm::Value {
        sea_orm::Value::String(Some(Box::new(self.0.to_owned())))
    }
}

impl ValueType for MyType {
    fn try_from(v: Value) -> std::result::Result<Self, ValueTypeErr> {
        match v {
            Value::String(Some(x)) => Ok(Self(*x)),
            _ => Err(ValueTypeErr),
        }
    }

    fn type_name() -> String {
        stringify!(MyType).to_owned()
    }

    fn column_type() -> ColumnType {
        ColumnType::String(None)
    }
}

try_getable_all!(MyType);
```

Though the macro is private.

Any advice? Thanks!

**Comment by billy1624** [comment, marked as answer]
Hey @tasn, welcome! You can find the links below for reference:
- https://github.com/SeaQL/sea-orm/pull/324 ([check this code snippet](https://github.com/SeaQL/sea-orm/pull/324/files#diff-8e6020819859a5806df7117b0d5cd5d56b84c63fb774597a7371076a8854e9e5))
- https://github.com/SeaQL/sea-orm/issues/116

**Reply by tasn** [reply]
Thanks a lot!

**Reply by billy1624** [reply]
Welcome! Just ping us here or in Discord if you got any other questions :)

**Reply by tasn** [reply]
Already got very very deep into the sea-orm code since asking that, so I'm good for now, but I'll let you know. Thanks!

**Reply by billy1624** [reply]
Nice!
